widget/completionentry.go

```diff
@@ -0,0 +1,203 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// CompletionEntry is an Entry with options displayed in a PopUpMenu.
+type CompletionEntry struct {
+	widget.Entry
+	popupMenu     *widget.PopUp
+	navigableList *navigableList
+	Options       []string
+	pause         bool
+	itemHeight    float32
+}
+
+// NewCompletionEntry creates a new CompletionEntry which creates a popup menu that responds to keystrokes to navigate through the items without losing the editing ability of the text input.
+func NewCompletionEntry(options []string) *CompletionEntry {
+	c := &CompletionEntry{Options: options}
+	c.ExtendBaseWidget(c)
+	return c
+}
+
+// ShowCompletion displays the completion menu
+func (c *CompletionEntry) ShowCompletion() {
+	if c.pause {
+		return
+	}
+	if len(c.Options) == 0 {
+		c.HideCompletion()
+		return
+	}
+
+	if c.navigableList == nil {
+		c.navigableList = newNavigableList(c.Options, &c.Entry, c.setTextFromMenu, c.HideCompletion)
+	}
+	holder := fyne.CurrentApp().Driver().CanvasForObject(c)
+
+	if c.popupMenu == nil {
+		c.popupMenu = widget.NewPopUp(c.navigableList, holder)
+	}
+	c.popupMenu.Resize(c.maxSize())
+	c.popupMenu.ShowAtPosition(c.popUpPos())
+	holder.Focus(c.navigableList)
+}
+
+// HideCompletion hides the completion menu.
+func (c *CompletionEntry) HideCompletion() {
+	if c.popupMenu != nil {
+		c.popupMenu.Hide()
+	}
+}
+
+// SetOptions set the completion list with itemList and update the view.
+func (c *CompletionEntry) SetOptions(itemList []string) {
+	c.Options = itemList
+	c.Refresh()
+}
+
+// Refresh the list to update the options to display.
+func (c *CompletionEntry) Refresh() {
+	c.Entry.Refresh()
+	if c.navigableList != nil {
+		c.navigableList.SetOptions(c.Options)
+	}
+}
+
+// calculate where the popup should appear
+func (c *CompletionEntry) popUpPos() fyne.Position {
```

Comment:
Please sort the methods alphabetically, with exported first then private, more info at https://github.com/fyne-io/fyne/wiki/Contributing

Reply:
@andydotxyz My editor uses `goimports` as explained in the contribution guide but the methods are not ordered.
This should happen because the methods you point are private I guess. So I will do it manually...

Reply:
I made changes, can you tell me if it's OK now ?

Reply:
Thanks. Sadly goimports does not fix the ordering of methods. 
We're working on it ;)